presto-main/src/main/java/com/facebook/presto/security/FileBasedSystemAccessControl.java

```diff
@@ -78,14 +80,36 @@ public SystemAccessControl create(Map<String, String> config)
             checkState(
                     configFileName != null,
                     "Security configuration must contain the '%s' property", CONFIG_FILE_NAME);
-
-            try {
-                Path path = Paths.get(configFileName);
-                if (!path.isAbsolute()) {
-                    path = path.toAbsolutePath();
+            final Path path = Paths.get(configFileName);
+            return RefreshingSystemAccessControl.optionallyRefresh(config, new Supplier<Optional<SystemAccessControl>>()
+            {
+                private long lastModifiedTime = -1;
+                @Override
+                public Optional<SystemAccessControl> get()
+                {
+                    Path toRead = path;
+                    if (!toRead.isAbsolute()) {
+                        toRead = path.toAbsolutePath();
+                    }
+                    // ensure we can read the file
+                    File file = toRead.toFile();
+                    // NOTE: we don't check the return value -- legacy behavior
+                    file.canRead();
+
+                    // check to see if the file has been updated, in which case attempt to load it
+                    long lastModified = file.lastModified();
+                    if(lastModified > lastModifiedTime){
+                        lastModifiedTime = lastModified;
+                        return Optional.of(load(toRead));
+                    }
+                    return Optional.empty();
```

Comment:
I would prefer to return previous value instead of empty here

Reply:
But refreshing framework handles not updates in case of empty. Cleaner in that the factory doesn't need to keep track of the previous either

Reply:
What is the difference between storing `lastModifiedTime` and previous value?

Reply:
It implies keeping track of two things - modified time and the last value. Here you only need to keep track of "do i need to create a new one"? If not, don't do anything.

Means you don't need to track another reference. I see both as equivalently complex to a user.